lib/ansible/modules/source_control/github_organization.py

```diff
@@ -0,0 +1,308 @@
+  returned: always
+  type: int
+  sample: 0
+private_repo_count:
+  description: The count of private repositories in the organization
+  returned: always
+  type: int
+  sample: 1
+default_repo_permission:
+  description: The default permission for repositories in the organization
+  returned: always
+  type: string
+  sample: read
+'''
+
+EXAMPLES = '''
+- name: Create an Organization
+  github_organization:
+    state: present
+    base_url: https://github-enterprise.domain.com
+    token: 'xxxxxxxxxxxxxxxxxxxxxxxxxxx'
+    token_user: token-username
+    name: my-organization
+    owner: a-user
+
+- name: Rename an organization
+  github_organization:
+      state: rename
+      base_url: https://github-enterprise.domain.com
+      token: 'xxxxxxxxxxxxxxxxxxxxxxxxxxx'
+      token_user: token-username
+      name: my-organization
+      rename_to: my-renamed-organization
+'''
+
+import base64
+from ansible.module_utils.urls import fetch_url
+from ansible.module_utils.basic import AnsibleModule
+
+
+def _build_api_url(base_url, path):
+    url = base_url
+
+    if url.endswith('/'):
+        if path.startswith('/'):
+            url += path[1:]
+        else:
+            url += path
+    else:
+        if path.startswith('/'):
+            url += path
+        else:
+            url += '/' + path
+
+    return url
+
+
+def _get_auth_header(user, token):
+    auth = base64.encodestring('%s:%s' % (user, token)).replace('\n', '')
+    return 'Basic %s' % auth
+
+
+def _get_organization(module, base_url, auth_header, name):
+    path = "orgs/%s" % (name)
+    url = _build_api_url(base_url, path)
+    headers = {'Authorization': auth_header}
+
+    r, info = fetch_url(module, url, method="GET", headers=headers)
+    if info['status'] == 200:
+        return module.from_json(r.read())
+    else:
+        return None
+
+
+def _get_base_url(module):
+    url = module.params['base_url']
+
+    if not url.endswith('api/v3'):
+        url = _build_api_url(url, 'api/v3')
+
+    return url
+
+
+def _merge_organization_result(res, org):
+    res.update({
+        'name': org['name'],
+        'id': org['id'],
+        'created_at': org['created_at'],
+        'updated_at': org['updated_at'],
+        'collaborators': org['collaborators'],
+        'public_repo_count': org['public_repos'],
+        'private_repo_count': org['total_private_repos'],
+        'default_repo_permission': org['default_repository_permission']
+    })
+    return res
+
+
+def create_organization(module, auth_header):
+    result = {'changed': False}
+
+    organization_name = module.params['name']
+    base_url = _get_base_url(module)
+
+    existing_org = _get_organization(module, base_url, auth_header, organization_name)
+
+    if existing_org is None:
+        url = _build_api_url(base_url, 'admin/organizations')
+        headers = {'Authorization': auth_header}
+        data = {
+            "login": organization_name,
+            # "profile_name": organization_name,
```

Comment:
Since having a display name is optional, you can update data if organization_name is provided.

Actually - since you've made name a required module option, you can just uncomment this line, right?

Reply:
The profile name is more like a description once the organisation is created, it is a fairly badly named parameter in the API. This corresponds to a "Organization display name" within the settings for an organization.

Most users (myself very much included) would likely prefer to leave this value unset, as not setting this value just uses the organization name.

I left it here just in case it may be desired in the future to add it back in via a parameter.